docs/reference/data-management/migrate-index-allocation-filters.asciidoc

```diff
@@ -0,0 +1,137 @@
+[role="xpack"]
+[[migrate-index-allocation-filters]]
+== Migrate index allocation filters to node roles
+
+If you currently use custom node attributes and
+<<shard-allocation-filtering, attribute-based allocation filters>> to
+move indices through <<data-tiers, data tiers>> in a hot-warm-cold architecture,
+we recommend that you switch to using the built-in node roles
+and automatic <<data-tier-shard-filtering, data tier allocation>>.
+Using node roles enables {ilm-init} to automatically
+move indices between data tiers.
+
+NOTE: While we recommend relying on automatic data tier allocation to manage
+your data in a hot-warm-cold architecture,
+you can still use attribute-based allocation filters to
+control shard allocation for other purposes.
+
+To switch to using node roles:
+
+. <<assign-data-tier, Assign data nodes>> to the appropriate data tier.
+. <<remove-custom-allocation-settings, Remove the attribute-based allocation
+settings>> from your {ilm} policy.
+. <<stop-setting-custom-hot-attribute, Stop setting the custom hot attribute>>
+on new indices.
+. Update existing indices to <<set-tier-preference, set a tier preference>>.
+
+
+[discrete]
+[[assign-data-tier]]
+=== Assign data nodes to a data tier
+
+Configure the appropriate roles for each data node to assign it to one or more
+data tiers: `data_hot`, `data_content`, `data_warm`, `data_cold`, or `data_frozen`.
+A node can also have other <<modules-node,roles>>. By default, new nodes are
+configured with all roles.
+
+When you add a data tier to an {ess} deployment,
+new nodes are automatically assigned to the tier.
+To change the role of an existing node in an {ess} deployment, use the
+{cloud}/ec-api-deployment-crud.html#ec_update_a_deployment[Update deployment API].
+Replace the node's `node_type` configuration with the appropriate `node_roles`.
+For example, the following configuration adds the node to the hot and content
+tiers, and enables it to act as an ingest node, remote, and transform node.
+
+[source,yaml]
+----
+"node_roles": [
+  "data_hot",
+  "data_content",
+  "ingest",
+  "remote_cluster_client",
+  "transform"
+],
+----
+
+If you are directly managing your own cluster,
+configure the appropriate roles for each node in `elasticsearch.yml`.
+For example, the following setting configures a node to be a data-only
+node in the hot and content tiers.
+
+[source,yaml]
+----
+node.roles [ data_hot, data_content ]
+----
+
+[discrete]
+[[remove-custom-allocation-settings]]
+=== Remove custom allocation settings from your {ilm-init} policy
+
+Update the allocate action for each phase to remove the attribute-based
+allocation settings. If the allocate action does not set the number of replicas,
+remove the allocate action entirely. (An empty allocate action is invalid.)
```

Comment:
Should we provide just a bit more context here? 

Namely, it's a bit unclear to me which policy are we talking about. If we expand and slightly duplicate the automatic data tier migration text here to reinforce the general goal, maybe it'll be a bit clearer we're talking about all _existing_ policies?

I'm thinking something along the lines of:
```
=== Remove custom allocation settings from your existing {ilm-init} policies

Update the allocate action for each phase to remove the attribute-based
allocation settings. This will enable the migrate action to be injected
in each ILM phase and allow {ilm-init} to automatically transition the indices
through the data tiers.

If the allocate action does not set the number of replicas,
remove the allocate action entirely (an empty allocate action is invalid)
```

What do you think?